packages/ember-htmlbars/tests/helpers/component_test.js

```diff
@@ -1,398 +0,0 @@
-import buildOwner from 'container/tests/test-helpers/build-owner';
-import { OWNER } from 'container/owner';
-
-var view, owner;
-
-QUnit.module('ember-htmlbars: {{#component}} helper', {
-  setup() {
-    owner = buildOwner();
-
-    owner.registerOptionsForType('template', { instantiate: false });
-    owner.register('component-lookup:main', ComponentLookup);
-  },
-
-  teardown() {
-    runDestroy(view);
-    runDestroy(owner);
-    owner = view = null;
-  }
-});
-
-QUnit.test('component helper with bound properties are updating correctly in init of component', function() {
-  owner.register('component:foo-bar', Component.extend({
-    init: function() {
-      this._super(...arguments);
-
-      equal(get(this, 'location'), 'Caracas', 'location is bound on init');
-    }
-  }));
-  owner.register('component:baz-qux', Component.extend({
-    init: function() {
-      this._super(...arguments);
-
-      equal(get(this, 'location'), 'Loisaida', 'location is bound on init');
-    }
-  }));
-  owner.register('template:components/foo-bar', compile('yippie! {{location}} {{yield}}'));
-  owner.register('template:components/baz-qux', compile('yummy {{location}} {{yield}}'));
-
-  view = EmberView.extend({
-    [OWNER]: owner,
-    dynamicComponent: computed('location', function() {
-      var location = get(this, 'location');
-
-      if (location === 'Caracas') {
-        return 'foo-bar';
-      } else {
-        return 'baz-qux';
-      }
-    }),
-    location: 'Caracas',
-    template: compile('{{#component view.dynamicComponent location=view.location}}arepas!{{/component}}')
-  }).create();
-
-  runAppend(view);
-  equal(view.$().text(), 'yippie! Caracas arepas!', 'component was looked up and rendered');
-
-  run(function() {
-    set(view, 'location', 'Loisaida');
-  });
-  equal(view.$().text(), 'yummy Loisaida arepas!', 'component was updated and re-rendered');
-
-  run(function() {
-    set(view, 'location', 'Caracas');
-  });
-  equal(view.$().text(), 'yippie! Caracas arepas!', 'component was updated up and rendered');
-});
-
-QUnit.test('component helper with unquoted string is bound', function() {
-  owner.register('template:components/foo-bar', compile('yippie! {{attrs.location}} {{yield}}'));
-  owner.register('template:components/baz-qux', compile('yummy {{attrs.location}} {{yield}}'));
-
-  view = EmberView.create({
-    [OWNER]: owner,
-    dynamicComponent: 'foo-bar',
-    location: 'Caracas',
-    template: compile('{{#component view.dynamicComponent location=view.location}}arepas!{{/component}}')
-  });
-
-  runAppend(view);
-  equal(view.$().text(), 'yippie! Caracas arepas!', 'component was looked up and rendered');
-
-  run(function() {
-    set(view, 'dynamicComponent', 'baz-qux');
-    set(view, 'location', 'Loisaida');
-  });
-  equal(view.$().text(), 'yummy Loisaida arepas!', 'component was updated and re-rendered');
-});
-
-QUnit.test('component helper destroys underlying component when it is swapped out', function() {
```

Comment:
moved to `@test component helper destroys underlying component when it is swapped out`